price-reporter/Dockerfile

```diff
@@ -11,39 +12,44 @@ RUN cat rust-toolchain | xargs rustup toolchain install
 # Install cargo-chef
 RUN cargo install cargo-chef
 
-FROM chef AS planner
-COPY . .
-RUN cargo chef prepare --recipe-path recipe.json
+# === Sources === #
+FROM chef AS sources
+WORKDIR /build
+COPY ./Cargo.toml ./Cargo.lock ./
+COPY ./price-reporter ./price-reporter
 
+# === Builder === #
 FROM chef AS builder
-COPY --from=planner /build/recipe.json recipe.json
 
 # Install protoc, openssl, and pkg-config
 RUN apt-get update && \
-    apt-get install -y pkg-config && \
-    apt-get install -y protobuf-compiler && \
-    apt-get install -y libssl-dev && \
-    apt-get install -y libclang-dev
+    apt-get install -y pkg-config protobuf-compiler libssl-dev libclang-dev
 
 # Disable compiler warnings and enable backtraces for panic debugging
 ENV RUSTFLAGS=-Awarnings
 ENV RUST_BACKTRACE=1
+ENV CARGO_HTTP_CHECK_REVOKE=false
+
+COPY --from=sources /build /build
+WORKDIR /build
+
+# Update Cargo.toml to include only "price-reporter" in workspace members
+RUN sed -i '/members[[:space:]]*=[[:space:]]*\[/,/\]/c\members = ["price-reporter"]' Cargo.toml
+RUN cargo chef prepare --recipe-path recipe.json --bin renegade-price-reporter
 
 # Build only the dependencies to cache them in this layer
 RUN cargo chef cook --release --recipe-path recipe.json
 
-# Build the relayer
-COPY . .
-RUN cargo build --release
+COPY --from=sources /build/price-reporter /build/price-reporter
+WORKDIR /build
```

Comment:
are these lines necessary? we are still in the `builder` stage where we've already `COPY --from=sources /build /build` and are already in `WORKDIR /build` (lines 33-34)

Reply:
yeah because chef will clobber the sources